File: src/main/java/com/example/apicentrale/mapper/PlayerMapper.java
package com.example.apicentrale.mapper;

import com.example.apicentrale.client.dto.ExternalPlayerDto;
import com.example.apicentrale.model.Player;
import org.springframework.stereotype.Component;


@Component
public class PlayerMapper {
    public Player toEntity(ExternalPlayerDto dto) {
        Player player = new Player();
        player.setId(dto.getId());
        player.setName(dto.getName());
        player.setNumber(dto.getNumber());
        player.setPosition(dto.getPosition());
        player.setNationality(dto.getNationality());
        player.setAge(dto.getAge());
        player.setChampionship(dto.getChampionship());
        player.setScoredGoals(dto.getScoredGoals());
        player.setPlayingTimeValue(dto.getPlayingTime().getValue().intValue());
        player.setPlayingTimeUnit(dto.getPlayingTime().getDurationUnit());
        return player;
    }
}

File: src/main/java/com/example/apicentrale/mapper/PlayerMapper.kt
package com.example.apicentrale.mapper

import com.example.apicentrale.client.dto.ExternalPlayerDto
import com.example.apicentrale.model.Player
import org.springframework.stereotype.Component

@Component
class PlayerMapper {
    fun toEntity(dto: ExternalPlayerDto): Player {
        val player = Player()
        player.setId(dto.id)
        player.setName(dto.name)
        player.setNumber(dto.number)
        player.setPosition(dto.position)
        player.setNationality(dto.nationality)
        player.setAge(dto.age)
        player.setChampionship(dto.championship)
        player.setScoredGoals(dto.scoredGoals)
        player.setPlayingTimeValue(dto.playingTime.value.toInt())
        player.setPlayingTimeUnit(dto.playingTime.durationUnit)
        return player
    }
}